app/locale.js

```diff
@@ -0,0 +1,181 @@
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+'use strict'
+
+const L20n = require('l20n')
+const path = require('path')
+const ipcMain = require('electron').ipcMain
+
+var menuIdentifiers = () => {
+  return [
+    'downloadsManager',
+    'confirmClearPasswords',
+    'about',
+    'quit',
+    'addToReadingList',
+    'viewPageSource',
+    'copyImageAddress',
+    'openImageInNewTab',
+    'saveImage',
+    'copyLinkAddress',
+    'openInNewSessionTab',
+    'openInNewPrivateTab',
+    'openInNewTab',
+    'openAllInTabs',
+    'disableAdBlock',
+    'disableTrackingProtection',
+    'muteTab',
+    'unmuteTab',
+    'pinTab',
+    'unpinTab',
+    'deleteFolder',
+    'deleteBookmark',
+    'editFolder',
+    'editBookmark',
+    'unmuteTabs',
+    'muteTabs',
+    'addBookmark',
+    'addFolder',
+    'newTab',
+    'closeTab',
+    'bookmarkPage',
+    'openFile',
+    'openLocation',
+    'openSearch',
+    'importFrom',
+    'closeWindow',
+    'savePageAs',
+    'spreadTheWord',
+    'share',
+    'undo',
+    'redo',
+    'cut',
+    'copy',
+    'paste',
+    'pasteWithoutFormatting',
+    'delete',
+    'selectAll',
+    'findNext',
+    'findPrevious',
+    'file',
+    'edit',
+    'view',
+    'actualSize',
+    'zoomIn',
+    'zoomOut',
+    'toolbars',
+    'reloadPage',
+    'reloadTab',
+    'cleanReload',
+    'readingView',
+    'tabManager',
+    'textEncoding',
+    'toggleDeveloperTools',
+    'toggleBrowserConsole',
+    'toggleFullScreenView',
+    'home',
+    'back',
+    'forward',
+    'reopenLastClosedWindow',
+    'showAllHistory',
+    'clearHistory',
+    'bookmarks',
+    'addToFavoritesBar',
+    'window',
+    'minimize',
+    'zoom',
+    'selectNextTab',
+    'selectPreviousTab',
+    'moveTabToNewWindow',
+    'mergeAllWindows',
+    'downloads',
+    'history',
+    'bringAllToFront',
+    'help',
+    'sendUsFeedback',
+    'services',
+    'hideOthers',
+    'showAll',
+    'newPrivateTab',
+    'newSessionTab',
+    'newWindow',
+    'reopenLastClosedTab',
+    'print',
+    'findOnPage',
+    'checkForUpdates',
+    'preferences',
+    'bookmarksManager',
+    'importBookmarks',
+    'reportAnIssue',
+    'submitFeedback',
+    'bookmarksToolbar',
+    'bravery',
+    'replaceAds',
+    'blockAds',
+    'allowAdsAndTracking',
+    'block3rdPartyCookie',
+    'blockPopups',
+    'httpsEverywhere'
+  ]
+}
+
+var ctx = null
+var translations = {}
+var lang = 'en-US'
+
+// Return a translate token from cache or a placeholder
+// indicating that no translation is available
+exports.translation = (token) => {
+  if (translations[token]) {
+    return translations[token]
+  } else {
+    return `[${token.toUpperCase()}]`
+  }
+}
+
+exports.init = (language, cb) => {
+  // Default to noop callback
+  cb = cb || function () {}
+
+  lang = language
+
+  // Languages to support
+  const langs = [
+    { code: 'en-US' },
+    { code: 'nl-NL' },
+    { code: 'pt-BR' }
```

Comment:
missing: bn-BD

Would be good to have a `docs/localizationDevelopment.md` file or similar which explains some of the manual things we need to do relating to localization.  For example list localizing menu items, what you need to do when adding a new language. etc.